pkg/asset/agent/image/agentartifacts.go

```diff
@@ -64,19 +68,22 @@ func (a *AgentArtifacts) Generate(dependencies asset.Parents) error {
 	a.RendezvousIP = ignition.RendezvousIP
 	a.IgnitionByte = ignitionByte
 	a.ISOPath = baseIso.File.Filename
+
 	a.Kargs = kargs.KernelCmdLine()
 	if agentconfig.Config != nil {
 		a.BootArtifactsBaseURL = strings.Trim(agentconfig.Config.BootArtifactsBaseURL, "/")
 	}
 
-	agentTuiFiles, err := a.fetchAgentTuiFiles(agentManifests.ClusterImageSet.Spec.ReleaseImage, agentManifests.GetPullSecretData(), registriesConf.MirrorConfig)
-	if err != nil {
-		return err
-	}
+	if agentManifests.AgentClusterInstall.Spec.PlatformType != hiveext.ExternalPlatformType && agentClusterInstall.GetExternalPlatformName() != string(models.PlatformTypeOci) {
```

Comment:
nit: this can reduced to

```
if agentClusterInstall.GetExternalPlatformName() != string(models.PlatformTypeOci) {
```

Reply:
actually there's a bug as well, if we're going to keep both conditions the `&&` should be `||`, otherwise this will never be true

Reply:
right, thanks for catching this. Just one condition to check platformName is also sufficient. Updated accordingly.